modules/ding_seo/ding_seo.tokens.inc

```diff
@@ -0,0 +1,250 @@
+<?php
+
+/**
+ * Implements hook_token_info().
+ * 
+ * Custom tokens for JSON LD. 
+ * A new token type 'Ding Seo' is added to give easier access to the tokens from the UI.  
+ * Tokens is defined with this hook, content is added later with hook_tokens()
+ * 
+ */
+
+function ding_seo_token_info()
+{
+    $type = [
+        'name' => t('Ding Seo'),
+        'description' => t('Tokens for structured data as JSON LD.'),
+    ];
+
+    // Location tokens
+    $node['location_name'] = [
+        'name' => t("Location Name"),
+        'description' => t('The location is the '),
+    ];
+    $node['location_url'] = [
+        'name' => t("Location Url"),
+        'description' => t('location_url'),
+    ];
+    $node['postalCode'] = [
+        'name' => t("Postal Code"),
+        'description' => t('Postal Code'),
+    ];
+    $node['streetAddress'] = [
+        'name' => t("Street Address"),
+        'description' => t('Street Address'),
+    ];
+    $node['addressLocality'] = [
+        'name' => t("Address Locality"),
+        'description' => t('Address Locality'),
+    ];
+    $node['addressRegion'] = [
+        'name' => t("Address Region"),
+        'description' => t('Address Region'),
+    ];
+    $node['addressCountry'] = [
+        'name' => t("Address Country"),
+        'description' => t('Address Country'),
+    ];
+    $node['longitude'] = [
+        'name' => t("longitude"),
+        'description' => t('longitude'),
+    ];
+    $node['latitude'] = [
+        'name' => t("latitude"),
+        'description' => t('latitude'),
+    ];
+  
+    // Logo dimensions tokens
+    $node['logo_width'] = [
+        'name' => t("logo_width"),
+        'description' => t('logo_width'),
+    ];
+    $node['logo_height'] = [
+        'name' => t("logo_height"),
+        'description' => t('logo_height'),
+    ];
+
+    // Event data formatting
+    $node['event_startdate'] = [
+        'name' => t("Event Startdate"),
+        'description' => t('Event enddate in YYYY-MM-DD HH:MM::SS format'),
+    ];
+    $node['event_enddate'] = [
+        'name' => t("Event Enddate"),
+        'description' => t('Event enddate in YYYY-MM-DD HH:MM::SS format'),
+    ];
+    $node['event_free'] = [
+        'name' => t("Event Free Boolean"),
+        'description' => t('Boolean of whether the event is free or not '),
+    ];
+
+    // Language
+    $node['language'] = [
```

Comment:
Is there really no language token defined by default on node? 

It appears that [node defines a language token](https://api.drupal.org/api/drupal/modules%21node%21node.tokens.inc/function/node_token_info/7.x), but maybe that can't be used? Perhaps a comment about this?

Reply:
There language token is there to set language und as dk, as there is a lot nodes with undefined language. 

case 'language':
                    if ($data["node"]->language == "und") {
                        $replacements[$original] = "dk";
                    } else {
                        $replacements[$original] = $data["node"]->language;
                    }
                    break;

Reply:
But wont this override the language token defined by the node module and perhaps cause unexpected behavior elsewhere?

Reply:
Nevermind above comment: If we make a `ding_seo_node_tokens` group it wont be an issue.

Reply:
Ok